server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java

```diff
@@ -122,149 +142,143 @@ protected void masterOperation(
                     finalListener.onResponse(new CreateIndexResponse(false, false, indexName));
                 }
             }, finalListener::onFailure);
-            clusterService.submitStateUpdateTask(
-                "auto create [" + request.index() + "]",
-                new AckedClusterStateUpdateTask(Priority.URGENT, request, listener) {
+            AckedClusterStateUpdateTask clusterTask = new AckedClusterStateUpdateTask(Priority.URGENT, request, listener) {
```

Comment:
Lets add a TODO here to move this away from `AckedClusterStateUpdateTask` and use a custom task object.

Reply:
Done